src/Graphics/Image/Processing/Noise.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-} 
+
+module Graphics.Image.Processing.Noise where
+
+import Control.Monad (forM_)
+import Control.Monad.ST
+import System.Random
+
+import Prelude as P hiding (subtract)
+import Graphics.Image.Interface as I
+import Graphics.Image
+import Graphics.Image.Types as IP
+import Graphics.Image.ColorSpace (X)
+
+-- | Helper function for generating a list of random co-ordinates.
+randomCoords :: StdGen -> Int -> Int -> [(Int,Int)]
+randomCoords a width height = (rnx1, rny1) : randomCoords g2 width height
+  where
+    (rnx1, g1) = randomR (1, width) a
```

Comment:
Is it illegal to have (0, 0) as random coordinate? Otherwise it should be `randomR (0, width)` and `randomR (0, height)`

Reply:
Yes sir, I'll check that.